community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/expressions/ContainerIndex.scala

```diff
@@ -33,18 +33,23 @@ with CollectionSupport {
   def compute(value: Any, ctx: ExecutionContext)(implicit state: QueryState): Any = {
     value match {
       case IsMap(m) =>
-        val idx = CastSupport.castOrFail[String](index(ctx))
-        m(state.query).getOrElse(idx, null)
+        val item = index(ctx)
+        if (item == null) null
+        else m(state.query).getOrElse(CastSupport.castOrFail[String](item), null)
```

Comment:
I don't understand, it is `castOrFail` that gives the `NullPointerException`. I thought about adding the null check in `castOrFail` but we use it from a lot of places and not all places have the same `null` semantics.